### PR TITLE
Add Attachments Interface

### PR DESCRIPTION
**Pull request by davidsoderberg**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Add so attachment can be sent.
- **What is the current behavior?** (You can also link to an open issue here)
https://github.com/notifirehq/notifire/issues/21
- **What is the new behavior (if this is a feature change)?**
Attachment can be sent.

### CODE REVIEW

**Comment by scopsy** [comment]
Thinking out loud, what is the best way to incorporate this to all the current providers and what about providers that don't support attachments until we migrate them all. @ComBarnea @davidsoderberg

**Comment by davidsoderberg** [comment]
Can we implement a feature property that is default false and then in handler we check if that one is true otherwise we throw an error if there is any attachments in payload?

File: packages/core/src/lib/handler/sms.handler.ts [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
+  ChannelTypeEnum,
+  IMessage,
+  ITriggerPayload,
+} from '../template/template.interface';
 
 export class SmsHandler {
   constructor(private message: IMessage, private provider: ISmsProvider) {}
 
   async send(data: ITriggerPayload) {
+    data.$attachments = data.$attachments?.filter((item) =>
```
Comment (scopsy) [inline packages/core/src/lib/handler/sms.handler.ts:13]:
```suggestion
    const attachments = data.$attachments?.filter((item) =>
```
I'd prefer if we don't override function params and instead create new const for that here

File: packages/core/src/lib/handler/sms.handler.ts [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
+      item.channels?.includes(ChannelTypeEnum.SMS)
```
Comment (scopsy) [inline packages/core/src/lib/handler/sms.handler.ts:14]: if no channels passed we should behave as if all channels we're selected  here (same on email)

Comment (davidsoderberg) [reply packages/core/src/lib/handler/sms.handler.ts:14]: So logic should be something like this:
```
item.channels === undefined ||
        item.channels.length === 0 ||
        item.channels?.includes(ChannelTypeEnum.EMAIL)
```
Did I understand you correctly?

Comment (scopsy) [reply packages/core/src/lib/handler/sms.handler.ts:14]: `item.channels?.length ? item.channels?.includes(ChannelTypeEnum.EMAIL) : true`

Maybe something like this?

Comment (scopsy) [reply packages/core/src/lib/handler/sms.handler.ts:14]: And we can add a unit test for this it would be amazing 🚀

**Comment by davidsoderberg** [comment]
@scopsy, can we implement a feature property that is default false and then in handler we check if that one is true otherwise we throw an error if there is any attachments in payload?

**Comment by scopsy** [comment]
@davidsoderberg I don't think it's neceserry. We can just create another PR. Into this one with the provider changes. And then merge everything together

**Comment by scopsy** [comment]
@davidsoderberg I've changed the base of the repo to the `v0.3.0` and added the full task list at #21 for what we need to finish for this. I'll give a hand as well on this. To get this right we need to do some modifications and update the providers. We then can release it on v0.3.0. WDYT?

File: packages/core/src/lib/content/content.engine.ts [hunk 3]
```diff
@@ -1,8 +1,12 @@
 import * as Handlebars from 'handlebars';
+import { IAttachmentOptions } from '../..';
```
Comment (scopsy) [inline packages/core/src/lib/content/content.engine.ts:2]: @davidsoderberg Not sure if this import is ok 😅

Comment (davidsoderberg) [reply packages/core/src/lib/content/content.engine.ts:2]: Woops, auto import messed up 😞

Comment (scopsy) [reply packages/core/src/lib/content/content.engine.ts:2]: Happens to me all the time 😅

**Comment by davidsoderberg** [comment]
> @davidsoderberg I've changed the base of the PR to the `v0.3.0` branch and added the full task list at #21 for what we need to finish for this. I'll give a hand as well on it. To get this right we need to do some modifications and update the providers. We then can release it on v0.3.0. WDYT?

I have also realized that this PR creates demands existing providers, sounds great 😄 
I have a solution to warn user of Notifire that the current provider does not support attatchments... 
Do you like to see it or should we look into anther solution for that later? How should I show it to you?

**Comment by scopsy** [comment]
You can just add a snippet on the new issue I created for that if that is ok at #104  :) 

I’ll merge this one for now

**Pull request merged**